src/core/border_router/routing_manager.hpp

```diff
@@ -685,14 +685,16 @@ class RoutingManager : public InstanceLocator
     //------------------------------------------------------------------------------------------------------------------
     // Typedefs
 
-    using Option                 = Ip6::Nd::Option;
-    using PrefixInfoOption       = Ip6::Nd::PrefixInfoOption;
-    using RouteInfoOption        = Ip6::Nd::RouteInfoOption;
-    using RaFlagsExtOption       = Ip6::Nd::RaFlagsExtOption;
-    using RouterAdvert           = Ip6::Nd::RouterAdvert;
-    using NeighborAdvertMessage  = Ip6::Nd::NeighborAdvertMessage;
-    using NeighborSolicitMessage = Ip6::Nd::NeighborSolicitMessage;
-    using RouterSolicitMessage   = Ip6::Nd::RouterSolicitMessage;
+    using Option                = Ip6::Nd::Option;
+    using PrefixInfoOption      = Ip6::Nd::PrefixInfoOption;
+    using RouteInfoOption       = Ip6::Nd::RouteInfoOption;
+    using RaFlagsExtOption      = Ip6::Nd::RaFlagsExtOption;
+    using RouterAdvert          = Ip6::Nd::RouterAdvert;
+    using NeighborAdvertMessage = Ip6::Nd::NeighborAdvertMessage;
+    using TxMessage             = Ip6::Nd::TxMessage;
+    using NeighborSolicitHeader = Ip6::Nd::NeighborSolicitHeader;
+    using RouterSolicitHeader   = Ip6::Nd::RouterSolicitHeader;
+    using LinkLayerAddress      = ot::BorderRouter::InfraIf::LinkLayerAddress;
```

Comment:
I think we can remove `ot::BorderRouter::` as we are already within the same name spaces in this file?
```suggestion
    using LinkLayerAddress      = InfraIf::LinkLayerAddress;
```